django/views/generic/edit.py

```diff
@@ -254,15 +254,18 @@ class DeletionMixin:
     """
     success_url = None
 
+    def delete_object(self):
+        success_url = self.get_success_url()
+        self.object.delete()
+        return HttpResponseRedirect(success_url)
+
     def delete(self, request, *args, **kwargs):
         """
         Calls the delete() method on the fetched object and then
         redirects to the success URL.
         """
         self.object = self.get_object()
-        success_url = self.get_success_url()
-        self.object.delete()
-        return HttpResponseRedirect(success_url)
+        return self.delete_object
```

Comment:
@auvipy 
I suppose the correct would be to execute the method `delete_object`, and not return it

```python
return self.delete_object()
```
and not

```python
return self.delete_object
```